cmd/create/execution.go

```diff
@@ -140,9 +143,16 @@ func createExecutionCommand(ctx context.Context, args []string, cmdCtx cmdCore.C
 	default:
 		return fmt.Errorf("invalid execution type %v", execParams.execType)
 	}
-	exec, _err := cmdCtx.AdminClient().CreateExecution(ctx, executionRequest)
-	if _err != nil {
-		return _err
+	var callOptions []grpc.CallOption
+	var exec *admin.ExecutionCreateResponse
+	grpcAPICall := func(_ctx context.Context, _callOptions []grpc.CallOption) error {
+		var err error
+		exec, err = cmdCtx.AdminClient().CreateExecution(_ctx, executionRequest, _callOptions...)
+		return err
+	}
+	err = auth.Do(ctx, cmdCtx.AuthClient(), grpcAPICall, callOptions)
```

Comment:
Is there a way to simplify this?
Like [here](https://github.com/flyteorg/flyteidl/pull/155/files#diff-017baee1fb74ab319b7caf1ef76d1321474dd5aec808c34b0a0f4f98b18c6878R179-R203)
You can build the authmetadataclient without initializing that singleton connection, then call FlyteClient() and if that returns then you can setup a PerCallRPCCredentials and set it as a dial option (instead of a call option) ... and therefore, you do not need to surround every single call to admin with this...

Reply:
Done. Created new MR for this https://github.com/flyteorg/flyteidl/pull/156/

Reply:
Will be updating this MR with just changes to flyteidl and couple of modules to be used. Rest there are no changes in flytectl side with this new movement